Leaf/jack/login authentication

misslame opened the pull request:
- added property `last_login` to be able to use class Account with Django login and auth feature
- Set up cookies and Authentication Backend to handle login, logout, and session requests
- change new login route `/auth/login/` in the frontend

blu3eee committed:
#42 login authentication

- added property `last_login` to be able to use class Account with Django login and auth feature
- change new login route `/auth/login/` in the frontend

blu3eee committed:
#335 remove old login views

shotoh commented on mesh/auth/views.py:
```diff
@@ -0,0 +1,83 @@
+    except Exception as e:
+        return JsonResponse({"error": str(e)}, status=500)
+
+def logout_view(request):
```
should there be a decorator requiring only POST requests?

shotoh commented on mesh/auth/backend.py:
```diff
@@ -0,0 +1,102 @@
+        """
+        from django.db import transaction
+        from mesh.accountSettings.models import Settings
+        with transaction.atomic():
```
if you want I believe you can take this out and use the @transaction.atomic decorator, up to you

shotoh reviewed (approved):
looks good, I left a few comments but I'm just nitpicking at this point

blu3eee committed:
#339 require POST method for logout request

add a decorator to limit the `logout` request to POST method only
https://github.com/LetsMesh/Site/pull/339#pullrequestreview-2069709605

blu3eee committed:
#339 use `transaction.atomic` instead

https://github.com/LetsMesh/Site/pull/339#pullrequestreview-2069709605

blu3eee committed:
#42 add `/auth` to app routes
